packages/cli/src/util/createServer.ts

```diff
@@ -97,10 +108,25 @@ function pipeOutputToSignale(stream: Readable) {
   });
 }
 
-export type CreatePrismOptions = {
+function isProxyServerOptions(options: CreateBaseServerOptions): options is CreateProxyServerOptions {
+  return !options.dynamic && 'upstream' in options;
```

Comment:
```suggestion
  return 'upstream' in options;
```

`options.dynamic = false` would be also truthy here. I think to check for upstream existence is enough.